### PR TITLE
removed usage of `table.getn` and `string.len`

### PR DESCRIPTION
**Pull request by Tieske**
### Summary

as per title

### Issues resolved

Fixes my urge to grab an axe and grep a hitlist from git blame

### CODE REVIEW

**Comment by Tieske** [comment]
tests pass locally

File: kong/plugins/basic-auth/access.lua [hunk 1]
```diff
@@ -34,7 +35,7 @@ local function retrieve_credentials(request, header_name, conf)
       return
     end
 
-    if m and table.getn(m) > 0 then
+    if next(m or empty) then
```
Comment (thibaultcha) [inline kong/plugins/basic-auth/access.lua]: We could avoid the empty table local with just:
```lua
if m and next(m) then
```

File: kong/plugins/oauth2/access.lua [hunk 2]
```diff
@@ -236,7 +238,7 @@ local function retrieve_client_credentials(parameters)
       return
     end
 
-    if m and table.getn(m) > 0 then
+    if next(m or empty) then
```
Comment (thibaultcha) [inline kong/plugins/oauth2/access.lua]: ditto

**Comment by Tieske** [comment]
updated as per comments

**Pull request merged**